KnowYourEnemyMutagen/Program.cs

```diff
@@ -179,16 +185,16 @@ private static void RunPatch(SynthesisState<ISkyrimMod, ISkyrimModGetter> state)
                     foreach (var eff in s.Effects)
                     {
                         eff.BaseEffect.TryResolve(state.LinkCache, out var baseEffect);
-                        if (baseEffect?.EditorID == null 
-                            || !resistancesAndWeaknesses.Contains(baseEffect.EditorID) 
+                        if (baseEffect?.EditorID == null
+                            || !resistancesAndWeaknesses.Contains(baseEffect.EditorID)
                             || eff.Data == null) continue;
                         var currentMagnitude = eff.Data.Magnitude;
                         eff.Data.Magnitude = AdjustMagicResist(currentMagnitude, effectIntensity);
-                        state.PatchMod.Spells.GetOrAddAsOverride(s);
+                        state.PatchMod.Spells.Add(s);
```

Comment:
So in the other usages, the DeepCopy() + late add wasn't adding much, as there was no short circuiting that would ever keep the Add from happening.   This was one case where it was being made use of, so was kept.

Just wanted to note on the GetOrAddAsOverride call, though.  That's going to do a query on the patch mod, and return the existing record if it already exists.  If it doesn't it will deep copy (again), insert it, then return it.

This isn't desirable for a few reasons:
1)  Extra deep copy for no reason
2)  Potential for your modified record to not be added!
      - You pass over spells, add a copy of each with some modifications
      - You pass over the spells again, doing a deep copy, modify the copy, then GetOrAddAsOverride
      - However, a record copy already exists in the mod (from the first pass), so GetOrAddAsOverride will just return it, without even looking/touching your copied and modified version.

Set is the proper call here, as you want it to insert into the group, even if you're replacing an existing one

Reply:
Add is also on the list of viable options, but it will throw if the key already exists, so Set is your boi